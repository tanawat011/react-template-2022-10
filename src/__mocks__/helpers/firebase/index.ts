import type * as Firebase from 'helpers/firebase'

const firebase = jest.createMockFromModule<typeof Firebase>('helpers/firebase')

module.exports = firebase
