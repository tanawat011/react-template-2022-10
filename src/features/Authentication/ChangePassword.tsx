export const ChangePassword = () => {
  return <>ChangePassword</>
}
