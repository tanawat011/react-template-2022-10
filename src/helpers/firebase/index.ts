export * from './deleteCollection'
export * from './deleteDocument'
export * from './getAllDocuments'
export * from './getDocument'
export * from './getDocumentWithRef'
export * from './setDocument'
export * from './subscribeCollection'
export * from './subscribeDocument'
export * from './updateDocument'
