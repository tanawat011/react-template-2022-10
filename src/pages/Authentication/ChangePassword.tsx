import { PATH } from 'Routes'
import { Button } from 'components/Button'

export const ChangePassword = () => {
  return (
    <>
      <div id='change-password-form'>
        <h1>Change Password</h1>

        <form className='flex flex-col'>
          <label id='password'>Old Password</label>
          <input id='password' />
          <label id='newPassword'>New Password</label>
          <input id='newPassword' />

          <div className='flex'>
            <Button link to={`../${PATH.AUTH.LOGIN}`}>
              Submit
            </Button>
          </div>
        </form>
      </div>
    </>
  )
}
