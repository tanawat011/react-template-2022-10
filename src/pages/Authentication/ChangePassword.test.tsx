import { renderWithProviders } from 'helpers/test'

import { ChangePassword } from './ChangePassword'

describe('<ChangePassword />', () => {
  test('match snapshot', () => {
    const { asFragment } = renderWithProviders(<ChangePassword />)
    expect(asFragment()).toMatchSnapshot()
  })
})
