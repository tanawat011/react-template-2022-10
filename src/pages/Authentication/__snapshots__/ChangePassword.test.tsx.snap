// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePassword /> match snapshot 1`] = `
<DocumentFragment>
  <div
    id="change-password-form"
  >
    <h1>
      Change Password
    </h1>
    <form
      class="flex flex-col"
    >
      <label
        id="password"
      >
        Old Password
      </label>
      <input
        id="password"
      />
      <label
        id="newPassword"
      >
        New Password
      </label>
      <input
        id="newPassword"
      />
      <div
        class="flex"
      >
        <button
          data-testid="button"
          to="../login"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
